services/base.js

```diff
@@ -165,6 +165,19 @@ class BaseService {
     return new RegExp(fullRegex)
   }
 
+  static get _cacheLength() {
+    if (this.category === 'build') {
```

Comment:
Would it be more readable to have a map category -> cache length? Keeping on adding `if` statements as we support more categories may become unpractical. `get _cacheLength()` would simply do a lookup in that map.